.github/workflows/ohos.yml

```diff
@@ -151,12 +151,77 @@ jobs:
       dromaeo: false
     secrets: inherit
 
+  # Generate a unique id that allows the timeout
+  # jobs to find the workload job run (via the job’s friendly name), even
+  # if there are multiple instances in the workflow call tree.
+  gen-uuids:
+    name: Generate unique runner IDs
+    runs-on: ubuntu-latest
+    outputs:
+      # re-use the same uuid for build, test, bench, with a prefix
+      build-unique-id: build-${{ steps.uuid.outputs.unique_id }}
+      test-unique-id: test-${{ steps.uuid.outputs.unique_id }}
+      bench-unique-id: bench-${{ steps.uuid.outputs.unique_id }}
+    steps:
+      - name: Generate a UUID
+        id: uuid
+        run: |
+          set -euo pipefail
+          unique_id=$(uuidgen)
+          echo "unique_id=$unique_id" | tee -a $GITHUB_OUTPUT
+
+  timeout-build:
+    name: Timeout for build-harmonyos-aarch64
+    needs:
+      - gen-uuids
+    env:
+      UNIQUE_ID: '${{ needs.gen-uuids.outputs.build-unique-id }}'
+      TIMEOUT: '600'
+      GH_TOKEN: '${{ secrets.GITHUB_TOKEN }}'
+      RUN_URL: '/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}'
+      SLEEP_PER_ITERATION: '15'
+    runs-on: ubuntu-latest
+    # Note: We copy-paste the steps here to avoid the 20 workflow limit of github.
+    # Todo: In the future we should check if we could move this to a composite-action.
```

Comment:
we can move it to composite action, as they only replace multiple steps (not jobs), so different limitations apply.